crates/holochain_sqlite/src/db/guard.rs

```diff
@@ -0,0 +1,54 @@
+use crate::db::conn::PConn;
+use crate::error::DatabaseResult;
+use rusqlite::Transaction;
+use std::ops::{Deref, DerefMut};
+use std::time::Instant;
+use tokio::sync::OwnedSemaphorePermit;
+
+pub(super) struct PConnGuard(PConn, OwnedSemaphorePermit);
+
+impl PConnGuard {
+    pub(super) fn new(conn: PConn, permit: OwnedSemaphorePermit) -> Self {
+        PConnGuard(conn, permit)
+    }
+}
+
+impl Deref for PConnGuard {
+    type Target = PConn;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for PConnGuard {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+/// Newtype to hand out connections that can only be used for running transactions
+pub struct PTxnGuard(PConnGuard, Instant);
```

Comment:
see above

```suggestion
pub(super) struct PTxnGuard(PConnGuard, Instant);
```

Reply:
This one is intentionally public, it's visible through the `get_read_txn` function of `DbRead`